app/src/Twig/AppExtension.php

```diff
@@ -100,4 +104,28 @@ public function messageReleaseToken(Msgs $message, User $user): string
     {
         return $this->messageService->getReleaseToken($message, $user);
     }
+        /**
+     * Get the original user when impersonating
+     * Returns the username of the original user, or empty string if not impersonating
+     */
+    public function getOriginalUser(): string
```

Comment:
Tant qu’à faire, on peut retourner le User directement, ou `null` s’il n’y a pas d’utilisateur impersonifié :


```suggestion
    public function getOriginalUser(): ?User
```